### PR TITLE
Fix Plugins links

### PR DESCRIPTION
**Pull request by MHBauer**
@moxiegirl 
These links do not work in my docs dev environment, so before these go out they need to be fixed.

### CODE REVIEW

**Comment by calavera** [comment]
LGTM. Thank you!

/cc @thaJeztah, @moxiegirl

**Comment by moxiegirl** [comment]
@MHBauer nice catch.  Can you tweak the index.md file for me? I didn't do a clean check there --- argh!

```
* [Understand Docker plugins](/extend/plugins)
* [Write a volume plugin](/extend/plugins_volume)
* [Docker plugin API](/extend/plugin_api)
```

**Comment by thaJeztah** [comment]
@moxiegirl just to prevent us going back and forth; Sven's PR looks to fix this in another way; https://github.com/docker/docker/pull/14668. I'm not sure his fix will solve the issue here as well (not tested yet on this PR), but we should have a clear plan/approach on how the links should be created.

(Preferably have a way to link between pages that also work when viewing the docs on GitHub, if possible)

**Comment by moxiegirl** [comment]
Sven is using rel and relref.  I haven't got consensus from the public docker repos yet on which coding to use. Until I get consensus, we should limit the use of the hugo code to the private repos.

**Comment by MHBauer** [comment]
Two or three things from my perspective:
1. I didn't originally change the index because those links worked.
2. the other links are still relative links, should I make those all the explicit links?
3. what's hugo? (and is there more info on docs that I should read up on? or a meeting to attend?)

**Comment by moxiegirl** [comment]
@MHBauer 
1. If you are testing your changes/what is already there with `make docs` and they work. Don't change em --- just note you tested.
2. See 2 above. :-D 
3. Hugo is a static site generator.  You can read up on it here: http://gohugo.io/   Not strictly necessary tho. :-D

File: docs/extend/index.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ weight = 6
 
 Currently, you can extend Docker by adding a plugin. This section contains the following topics:
 
-* [Understand Docker plugins](plugins.md)
-* [Write a volume plugin](plugins_volumes.md)
-* [Docker plugin API](plugin_api.md)
+* [Understand Docker plugins](/extend/plugins)
+* [Write a volume plugin](/extend/plugins_volumes)
```
Comment (moxiegirl) [inline docs/extend/index.md]: this should be `plugins_volume`

**Comment by moxiegirl** [comment]
@MHBauer Sorry for the confusion. With that one change LGTM

**Comment by MHBauer** [comment]
Change made. Everything is consistent and all the links work when I do `make docs`.

**Comment by thaJeztah** [comment]
LGTM thanks!

**Comment by srust** [comment]
Hi,

Did you intend for these docs also to be browsable through github? The links don't seem to work browsing through the source tree, such as from:

https://github.com/docker/docker/blob/master/docs/extend/index.md

Because the links don't have the ".md" in the paths, none of the links from index.md work for me in the browser. I'd be happy to submit a PR, but wanted to check first. Personally, I think it would be useful to have them work that way too, unless I'm missing something?

Thanks!

**Comment by thaJeztah** [comment]
@srust currently, the .md files are not intended to be read on GitHub, and only used to generate the online docs.

Changing the links would make them work on GitHub, but *break* https://docs.docker.com

We are aware of this issue and *hope* to find a way around this some day.

**Comment by srust** [comment]
Okay cool.  That's what I was suspecting.. Thanks for the info and the reply!

**Comment by thaJeztah** [comment]
@srust you're welcome! (and I fully agree that it's currently an inconvenience)

**Comment by thaJeztah** [comment]
ping @calavera for cherry pick

**Pull request merged**